Test/SyntaxSugar.hpp

```diff
@@ -683,14 +683,9 @@ inline Clause* clause(Stack<Lit> ls) {
         { return !l.selected(); })
     .unwrapOrElse( [&]() {return ls.size(); });
 
-  Clause& out = *new(ls.size()) Clause(ls.size(), testInf); 
+  Clause& out = *Clause::fromIterator(arrayIter(ls)
+      .map([](auto l) -> Literal* { return l; }), testInf);
```

Comment:
Isn't this just an identity `map`ping?

Reply:
Not quite. `auto l` is actually `Lit l`. so we're mapping `Lit -> Literal*`. I ll make this explicit.

Reply:
Cool, thanks!